pkg/fakerp/fakerp.go

```diff
@@ -118,7 +122,72 @@ func GetDeployer(log *logrus.Entry, cs *api.OpenShiftManagedCluster, testConfig
 			log.Warnf("deployment failed: %#v", err)
 			debugDeployerError(ctx, log, cs, err, testConfig)
 		}
-		return err
+
+		// This causes fails in our tests because deployer function is not versioned.
+		// TODO: Remove this when v7 is gone and PLS is default in the deployer function
+		if cs.Config.PluginVersion == "v9.0" &&
+			conf != nil {
+			log.Info("applying PLS deployment")
+			now := time.Now().Unix()
+			plsTemplate, err := arm.GenerateClusterSide(ctx, cs)
+			if err != nil {
+				return nil, err
+			}
+
+			future, err = deployments.CreateOrUpdate(ctx, cs.Properties.AzProfile.ResourceGroup, "plsdeploy", azresources.Deployment{
+				Properties: &azresources.DeploymentProperties{
+					Template: plsTemplate,
+					Mode:     azresources.Incremental,
+				},
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			log.Info("waiting for arm template deployment to complete")
+			err = future.WaitForCompletionRef(ctx, deployments.Client())
+			if err != nil {
+				log.Warnf("deployment failed: %#v", err)
+				debugDeployerError(ctx, log, cs, err, testConfig)
+			}
+
+			log.Info("applying PE deployment")
+			peTemplate, err := arm.GenerateRPSide(ctx, cs, conf, now)
+			if err != nil {
+				return nil, err
+			}
+
+			future, err = deployments.CreateOrUpdate(ctx, conf.ManagementResourceGroup, fmt.Sprintf("pedeploy-%d", now), azresources.Deployment{
+				Properties: &azresources.DeploymentProperties{
+					Template: peTemplate,
+					Mode:     azresources.Incremental,
+				},
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			log.Info("waiting for arm template deployment to complete")
+			err = future.WaitForCompletionRef(ctx, deployments.Client())
+			if err != nil {
+				log.Warnf("deployment failed: %#v", err)
+				debugDeployerError(ctx, log, cs, err, testConfig)
+			}
+
+			log.Info("get PE IP address")
+			nm, err := newNetworkManager(ctx, log, cs.Properties.AzProfile.SubscriptionID, conf.ManagementResourceGroup)
+			if err != nil {
+				return nil, err
+			}
+			peIP, err := nm.getPrivateEndpointIP(ctx, fmt.Sprintf("%s-%s-%d", utilarm.PrivateEndpointNamePrefix, cs.Name, now))
```

Comment:
should I also be expecting to see a call to getPrivateEndpointIP to populate the relevant field in the internal model in advance of every fakerp->plugin call?  That's what the real RP will be doing, right?

Reply:
(this could be part of the connect wrapper follow-up)

Reply:
Yes. But at the moment MSFT has their client and we do ours. This is pure technical depts after rendering..